libs/aws/langchain_aws/rerank/rerank.py

```diff
@@ -0,0 +1,142 @@
+from copy import deepcopy
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+import boto3
+from langchain_core.callbacks.manager import Callbacks
+from langchain_core.documents import BaseDocumentCompressor, Document
+from langchain_core.utils import from_env
+from pydantic import ConfigDict, Field, model_validator
+from typing_extensions import Self
+
+
+class BedrockRerank(BaseDocumentCompressor):
+    """Document compressor that uses AWS Bedrock Rerank API."""
+
+    client: Any = None
+    """Bedrock client to use for compressing documents."""
+    top_n: Optional[int] = 3
+    """Number of documents to return."""
+    model_id: Optional[str] = "amazon.rerank-v1:0"
+    """Model ID to fetch ARN dynamically."""
+    aws_region: str = Field(
+        default_factory=from_env("AWS_DEFAULT_REGION", default=None)
+    )
```

Comment:
Though I like the short name here, we have been using `region_name` in other places, and would prefer we keep it consistent here as well.
```suggestion
    region_name: str = Field(
        default_factory=from_env("AWS_DEFAULT_REGION", default=None)
    )
```

Reply:
Solved on 04891259cee476800dfe2de507faefc61ada9115